maven_iuvs/echelle.py

```diff
@@ -841,3 +880,1530 @@ def find_files_with_geometry(file_index):
+    mainax.set_title(t)
+
+    # Plot background fit
+    if plot_bg:
+        thebackground = background(data_wavelengths, fit_params_for_printing['M'], fit_params_for_printing['lambdac'], fit_params_for_printing['B'])
+        mainax.plot(data_wavelengths, thebackground, label="background", linewidth=2, zorder=2)
+        
+    # Plot the data and fit and a guideline for the central wavelength
+    mainax.errorbar(data_wavelengths, data_vals, yerr=data_unc, label="data", linewidth=1, zorder=3, alpha=0.7)
+    mainax.plot(data_wavelengths, model_fit, label="model", linewidth=2, zorder=2)
+    mainax.axvline(fit_params_for_printing['lambdac'], color="gray", zorder=1, linewidth=0.5, )
+
+    if show_integration_regions:
+        mainax.fill_betweenx([0, np.max(data_vals)], data_wavelengths[H_a], x2=data_wavelengths[H_b], color="xkcd:salmon", alpha=0.3, zorder=1)
+        mainax.fill_betweenx([0, np.max(data_vals)], data_wavelengths[D_a], x2=data_wavelengths[D_b], color="xkcd:slate blue", alpha=0.3, zorder=1)
+
+    # get index of lambda for D so we can find the value there
+    if fit_params_for_printing["lambdac_D"] is not np.nan:
+        D_i = find_nearest(data_wavelengths, fit_params_for_printing['lambdac_D'])[0]
+        mainax.axvline(fit_params_for_printing['lambdac_D'], color="gray", linewidth=0.5, zorder=1)
+    
+    # Print text
+    printme = [#r"H $\lambda_c$: "+f"{round(fit_params_for_printing['lambdac'], 3)}", 
+               #r"D $\lambda_c$: "+f"{round(fit_params_for_printing['lambdac_D'], 3)}",
+              ]
+    
+    # Now subtract the background entirely from the fit and then integrate to see the total brightness
+    if "kR" in unit:
+        # background_subtracted = data_vals - thebackground
+        # Htot, Dtot = line_brightness(data_wavelengths, background_subtracted, [H_a, H_b], [D_a, D_b])
+        printme.append(f"H: {fit_params_for_printing['area']} ± {fit_params_for_printing['uncert_H']} kR")
+        printme.append(f"D: {fit_params_for_printing['area_D']} ± {fit_params_for_printing['uncert_D']} kR")
+        textx = [0.38, 0.28]
+        texty = [0.5, 0.17]
+        talign = ["left", "right"]
+
+    for i in range(0, len(printme)):
+        mainax.text(textx[i], texty[i], printme[i], transform=mainax.transAxes, ha=talign[i])
+
+    # ax.set_yscale("log")
+    mainax.set_ylabel(f"Brightness ({unit})")
+    if logview:
+        mainax.set_yscale("log")
+    mainax.legend()
+    mainax.set_xlim(min(data_wavelengths)-0.02, max(data_wavelengths)+0.02)
+
+    # Residual axis
+    sign = np.sign(model_fit-data_vals)
+    residual = sign * (data_vals - model_fit)**2
+    residax.plot(data_wavelengths, residual, linewidth=1)
+    residax.set_ylabel(f"Residuals\n (sgn((data-model)^2))")
+    residax.set_xlabel("Wavelength (nm)")
+    
+    plt.show()
+
+    pass
+
+
+def plot_line_fit_comparison(data_wavelengths, data_vals_new, data_vals_BU, model_fit_new, model_fit_BU, fit_params_new, fit_params_BU, BUbackground, 
```

Comment:
should plotting functions be in echelle_graphics?

Reply:
Yes, definitely, but I left them here to avoid a circular import problem. Upon review, I think I can fix the problem so that they can be moved to the right place. Stand by.

Reply:
Fixed by [dd078e8](https://github.com/lasp/maven_iuvs/pull/66/commits/dd078e895dc4a009322e3eb6dbe764070be49269)